src/circuit_proof/assignment.rs

```diff
@@ -0,0 +1,135 @@
+use curve25519_dalek::scalar::Scalar;
+use errors::R1CSError;
+use std::ops::{Add, Div, Mul, Sub, Try};
+
+// The assignment value to a variable, as stored in `ConstraintSystem`.
+// Provers create a `Value` assignment, while verifiers create an `Missing` assignment.
+#[derive(Copy, Clone, Debug)]
+pub enum Assignment {
+    Value(Scalar),
+    Missing(),
+}
+
+impl Assignment {
+    pub fn new(scalar: Scalar) -> Self {
```

Comment:
Could this be `impl From<Scalar> for Assignment`?

Reply:
yeah!
I also implemented From<u64> while I was at it, thanks for the suggestion